app/models/qing_group_odk_partitioner.rb

```diff
@@ -0,0 +1,88 @@
+# Class responsible to reorganize the questions
+# It's main concern is to separate a qing group that contains multilevel
+# questionings into several groups. The idea here is to just remove the 
+# multilevel questioning from the group, since ODK doesn't show it correctly
+# if it's a child of a group.
+#
+# Input:
+# {
+#   Qing => {},
+#   QingGroup => {
+#     Qing(1) => {},
+#     Qing(2) => {},
+#     Qing(3-multilevel) => {},
+#     Qing(4) => {}
+#   },
+#   Qing => {},
+# }
+#
+# Output:
+# {
+#   Qing => {},
+#   QingGroup => {
+#     Qing(1) => {},
+#     Qing(2) => {},
+#   },
+#   Qing(3-multilevel) => {},
+#   QingGroup => {
+#     Qing(4) => {}
+#   },
+#   Qing => {},
+# }
+class QingGroupOdkPartitioner
+  attr_accessor :descendants, :organized_descendants
+
+  def initialize(descendants)
+    @descendants = descendants
+    @organized_descendants = {}
+  end
+
+  def organize
+    @descendants.each do |key, value|
+      if key.is_a? QingGroup
+        split_qing_group_as_necessary(key, value)
+      else
+        store_regular_qing(key)
+      end
+    end
+
+    @organized_descendants
+  end
+
+  private
+
+  def split_qing_group_as_necessary(group, questionings)
+    qing_group_divided = QingGroupDivided.new(group)
+
+    questionings.each do |qing|
+      qing_object = qing[0]
+
+      if (qing_object.multi_level?)
+        store_qing_group(qing_group_divided)
+        store_multilevel_qing_outside_a_group(qing_object);
+
+        # Start another group to separate these questionings from the old ones
+        qing_group_divided = QingGroupDivided.new(group)
+      else
+        qing_group_divided.children[qing_object] = ''
```

Comment:
I think a comment here is warranted. Setting the hash value to "" is not super intuitive.